In [1]:
from transformers import DistilBertTokenizer, DistilBertModel
import torch

tokenizer = DistilBertTokenizer.from_pretrained('distilbert-base-cased')
model = DistilBertModel.from_pretrained('distilbert-base-multilingual-cased')

input_ids = torch.tensor(tokenizer.encode("Hello, how are you?")).unsqueeze(0)
outputs = model(input_ids)
print(outputs["last_hidden_state"].shape)

Some weights of the model checkpoint at distilbert-base-multilingual-cased were not used when initializing DistilBertModel: ['vocab_projector.bias', 'vocab_transform.bias', 'vocab_layer_norm.bias', 'vocab_projector.weight', 'vocab_layer_norm.weight', 'vocab_transform.weight']
- This IS expected if you are initializing DistilBertModel from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing DistilBertModel from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).


torch.Size([1, 8, 768])


* Batch Size (1): Dies zeigt an, dass Ihre Eingabe nur aus einem einzigen Datensatz (oder Beispiel) besteht. In neuronalen Netzen wird oft eine Gruppe von Daten gleichzeitig verarbeitet (genannt "Batch"). Da Sie hier nur einen Satz verarbeiten, ist die Batch-Größe 1.

* Anzahl der Tokens (8): Dies ist die Anzahl der Tokens, in die Ihr Eingabetext zerlegt wurde. In Ihrem Fall wird der Satz "Hello, how are" vom Tokenizer in 6 Tokens aufgeteilt. Diese 6 Tokens beinhalten auch spezielle Tokens wie [CLS] (am Anfang) und [SEP] (am Ende), die vom BERT-Modell für verschiedene Zwecke verwendet werden. Der Tokenizer fügt diese speziellen Tokens automatisch hinzu.

* Embedding-Größe (768): Dies ist die Größe des Embedding-Vektors für jedes Token. DistilBERT, wie viele BERT-Varianten, verwendet 768-dimensionale Vektoren, um die Merkmale jedes Tokens darzustellen.